sw/host/opentitanlib/src/transport/verilator/uart.rs

```diff
@@ -0,0 +1,60 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::Result;
+use std::fs::File;
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::time::Duration;
+
+use crate::io::uart::Uart;
+use crate::util::file;
+
+/// Represents the verilator virtual UART.
+pub struct VerilatorUart {
+    file: File,
+}
+
+impl VerilatorUart {
+    pub fn open(path: &str) -> Result<Self> {
+        Ok(VerilatorUart {
+            file: OpenOptions::new().read(true).write(true).open(path)?,
+        })
+    }
+}
+
+impl Uart for VerilatorUart {
+    fn get_baudrate(&self) -> u32 {
+        // The verilator UART operates at 7200 baud.
+        // See `sw/device/lib/arch/device_sim_verilator.c`.
+        7200
+    }
+
+    fn set_baudrate(&mut self, _baudrate: u32) -> Result<()> {
+        // As a virtual uart, setting the baudrate is a no-op.
+        Ok(())
+    }
+
+    fn read_timeout(&mut self, buf: &mut [u8], timeout: Duration) -> Result<usize> {
+        file::wait_read_timeout(&self.file, timeout)?;
+        let n = self.file.read(buf)?;
```

Comment:
Inline n?

Reply:
There are a couple of options here:
1) as-is, I'm returning `Ok(n)`, which is the number of bytes read.
2) `Ok(self.file.read(buf)?)` is the same.  I'm relying on `?` to automatically deal with error type conversion.  Feels a bit weird to me to use `?` directly inside of `Ok(...)`.
3) `self.file.read(buf).map_err(|e| e.into())` to perform the explicit error type conversion.

Which way seems best to you?

Reply:
I've totally written #2 before. If you don't like it, we can keep it as-is. =)